examples/lfric/code/matrix_vector_kernel_mod.F90

```diff
@@ -109,24 +111,20 @@ subroutine matrix_vector_kernel_code(cell,              &
   integer(kind=i_def), dimension(ndf2), intent(in) :: map2
   real(kind=r_def), dimension(undf2),              intent(in)    :: x
   real(kind=r_def), dimension(undf1),              intent(inout) :: lhs
-  real(kind=r_def), dimension(ndf1,ndf2,ncell_3d), intent(in)    :: matrix
+  real(kind=r_def), dimension(ncell_3d,ndf1,ndf2), intent(in)    :: matrix
 
   ! Internal variables
-  integer(kind=i_def)               :: df, k, ik
-  real(kind=r_def), dimension(ndf2) :: x_e
-  real(kind=r_def), dimension(ndf1) :: lhs_e
+  integer(kind=i_def) :: df, df2, k, ik
 
-  do k = 0, nlayers-1
-    do df = 1, ndf2
-      x_e(df) = x(map2(df)+k)
-    end do
-    ik = (cell-1)*nlayers + k + 1
-    lhs_e = matmul(matrix(:,:,ik),x_e)
-    do df = 1,ndf1
-       lhs(map1(df)+k) = lhs(map1(df)+k) + lhs_e(df)
+  do df = 1, ndf1
+    do df2 = 1, ndf2
+      do k = 0, nlayers-1
+        ik = (cell-1)*nlayers + k + 1
```

Comment:
It can and I did that for this and all copies of the kernel. I made the same change in an LFRic branch and tested it, all good. I'm not sure whether we need to rush this change in LFRic for vn2.0, it can be done in slower time. It seems that there are lots of kernels in LFRic where we can reduce operations so it merits a further thought.